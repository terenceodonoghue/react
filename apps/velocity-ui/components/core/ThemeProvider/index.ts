export { default, defaultTheme } from './ThemeProvider';
export type { ThemeProviderProps } from './ThemeProvider';
