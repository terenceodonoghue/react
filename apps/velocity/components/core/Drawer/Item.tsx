import { useTheme } from '@emotion/react';
import { rgba, transitions } from 'polished';
import {
  forwardRef,
  JSXElementConstructor,
  LiHTMLAttributes,
  SVGProps,
} from 'react';

export interface ItemProps extends LiHTMLAttributes<HTMLLIElement> {
  compact?: boolean;
  icon?: JSXElementConstructor<SVGProps<SVGSVGElement>>;
  selected?: boolean;
}

const Item = forwardRef<HTMLLIElement, ItemProps>(
  (
    { children, compact = false, icon: Icon, selected = false, ...props },
    ref,
  ) => {
    const theme = useTheme();

    return (
      <li
        css={[
          ({ color, transition }) => ({
            alignItems: 'center',
            backgroundColor: selected ? rgba(color.primary, 0.1) : undefined,
            borderLeft: `solid 3px ${selected ? color.primary : 'transparent'}`,
            color: selected ? color.primary : color.neutral[500],
            cursor: 'pointer',
            display: 'flex',
            height: 56,
            justifyContent: 'flex-start',
            padding: '0 0 0 25px',
            ...transitions(['background-color'], transition.quickly),
            '&:hover': {
              backgroundColor: !selected && rgba(color.primary, 0.1),
            },
          }),
        ]}
        data-testid="drawer-item"
        ref={ref}
        {...props}
      >
        {Icon && (
          <Icon
            color={theme.color.neutral[400]}
            css={{
              flexShrink: 0,
              height: 24,
              width: 24,
            }}
            data-testid="drawer-item-icon"
          />
        )}
        <span
          css={[
            ({ transition }) => ({
              marginLeft: 16,
              opacity: compact ? 0 : 1,
              ...transitions(['opacity'], transition.quickly),
            }),
          ]}
          data-testid="drawer-item-text"
        >
          {children}
        </span>
      </li>
    );
  },
);

export default Item;
