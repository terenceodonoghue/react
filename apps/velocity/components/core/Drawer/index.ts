import Drawer from './Drawer';

export { default as DrawerItem } from './Item';
export default Drawer;
