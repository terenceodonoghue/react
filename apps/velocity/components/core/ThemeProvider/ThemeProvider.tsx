import { ThemeProvider as EmotionProvider, Theme } from '@emotion/react';
import merge from 'lodash/merge';
import { FunctionComponent, ReactNode } from 'react';

export const defaultTheme: Theme = {
  color: {
    primary: '#2e5bff',
    secondary: '#8c54ff',
    neutral: {
      50: '#fff',
      100: '#f4f6fc',
      200: '#eef3f5',
      300: '#e4e8f0',
      400: '#bfc5d2',
      500: '#b0bac9',
      600: '#8798ad',
      700: '#8097b1',
      800: '#69707f',
      900: '#2e384d',
    },
    ui: {
      blue: '#2e5bff',
      green: '#33ac2e',
      purple: '#8c54ff',
      red: '#d63649',
      teal: '#00c1d4',
      yellow: '#f7c137',
    },
    tonalOffset: 0.2,
  },
  font: {
    family: 'Rubik, sans-serif',
    weight: {
      light: 300,
      regular: 400,
      medium: 500,
    },
  },
  transition: {
    quickly: '250ms cubic-bezier(0.4, 0, 0.2, 1)',
    slowly: '300ms cubic-bezier(0.4, 0, 0.6, 1)',
  },
};

export interface ThemeProviderProps {
  children: ReactNode;
  theme?: Theme;
}

const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  theme,
  ...props
}) => (
  <EmotionProvider
    theme={theme ? merge(defaultTheme, theme) : defaultTheme}
    {...props}
  />
);

export default ThemeProvider;
