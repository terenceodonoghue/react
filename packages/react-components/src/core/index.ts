export { default as Flex } from './Flex.js';
export { default as Global } from './Global.js';
