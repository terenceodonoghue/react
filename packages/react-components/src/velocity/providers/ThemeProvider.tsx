import {
  css,
  Global,
  GlobalProps,
  ThemeProvider as EmotionProvider,
  Theme,
} from '@emotion/react';
import { merge } from 'lodash-es';
import { PropsWithChildren } from 'react';

export const defaultTheme: Theme = {
  color: {
    primary: '#2e5bff',
    secondary: '#e0e7ff',
    neutral: {
      50: '#fff',
      100: '#f4f6fc',
      200: '#eef3f5',
      300: '#e4e8f0',
      400: '#bfc5d2',
      500: '#b0bac9',
      600: '#8798ad',
      700: '#8097b1',
      800: '#69707f',
      900: '#2e384d',
    },
    ui: {
      blue: '#2e5bff',
      green: '#33ac2e',
      purple: '#8c54ff',
      red: '#d63649',
      teal: '#00c1d4',
      yellow: '#f7c137',
    },
    tonalOffset: 0.2,
  },
  font: {
    family: 'Rubik, sans-serif',
    weight: {
      light: 300,
      regular: 400,
      medium: 500,
    },
  },
  page: {
    maxWidth: 1360,
  },
  transition: {
    quickly: '250ms cubic-bezier(0.4, 0, 0.2, 1)',
    slowly: '300ms cubic-bezier(0.4, 0, 0.6, 1)',
  },
};

export interface ThemeProviderProps extends Partial<GlobalProps> {
  theme?: Theme;
}

const ThemeProvider = ({
  children,
  styles,
  theme,
}: PropsWithChildren<ThemeProviderProps>) => (
  <EmotionProvider theme={theme ? merge(defaultTheme, theme) : defaultTheme}>
    <Global
      styles={[
        css`
          @import url('https://unpkg.com/sanitize.css');
          @import url('https://fonts.googleapis.com/css2?family=Rubik:wght@300;400;500&display=swap');
        `,
        {
          ':any-link': {
            textDecoration: 'none',
          },
          '.recharts-surface': {
            overflow: 'visible',
          },
        },
        styles,
      ]}
    />
    {children}
  </EmotionProvider>
);

export default ThemeProvider;
