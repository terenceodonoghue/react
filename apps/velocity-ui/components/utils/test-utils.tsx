/* eslint-disable import/no-extraneous-dependencies */
import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider, {
  ThemeProviderProps,
} from '~/components/core/ThemeProvider';

const Provider: FunctionComponent<ThemeProviderProps> = ({ children }) => (
  <ThemeProvider>{children}</ThemeProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Provider, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
