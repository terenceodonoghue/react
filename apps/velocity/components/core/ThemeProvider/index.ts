import ThemeProvider, { defaultTheme } from './ThemeProvider';

export { defaultTheme };
export default ThemeProvider;
export type { ThemeProviderProps } from './ThemeProvider';
