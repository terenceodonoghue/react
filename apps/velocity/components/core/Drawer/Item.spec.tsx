import React from 'react';
import { faker } from '@faker-js/faker';
import { rgba } from 'polished';
import { render } from 'test-utils';
import { defaultTheme } from '~/components/core/ThemeProvider';
import Item from './Item';

describe('Item', () => {
  it('renders content', () => {
    const content = faker.hacker.phrase();
    const { container, getByTestId } = render(<Item>{content}</Item>);

    expect(getByTestId('drawer-item-text')).toHaveTextContent(content);
    expect(container).toMatchSnapshot();
  });

  it('shows text by default', () => {
    const { container, getByTestId } = render(<Item />);

    expect(getByTestId('drawer-item-text')).toHaveStyle({
      opacity: 1,
    });
    expect(container).toMatchSnapshot();
  });

  it('renders compact variant', () => {
    const { container, getByTestId } = render(<Item compact />);

    expect(getByTestId('drawer-item-text')).toHaveStyle({
      opacity: 0,
    });
    expect(container).toMatchSnapshot();
  });

  it('is unselected by default', () => {
    const { container, getByTestId } = render(<Item />);

    expect(getByTestId('drawer-item')).toHaveStyle({
      borderLeft: 'solid 3px transparent',
    });
    expect(getByTestId('drawer-item')).not.toHaveStyle({
      backgroundColor: rgba(defaultTheme.palette.accent, 0.1),
    });
    expect(container).toMatchSnapshot();
  });

  it('renders selected item', () => {
    const { container, getByTestId } = render(<Item selected />);

    expect(getByTestId('drawer-item')).toHaveStyle({
      borderLeft: `solid 3px ${defaultTheme.palette.accent}`,
    });
    expect(getByTestId('drawer-item')).toHaveStyle({
      backgroundColor: rgba(defaultTheme.palette.accent, 0.1),
    });
    expect(container).toMatchSnapshot();
  });

  it('renders an icon', () => {
    const { container, getByTestId } = render(
      <Item icon={(props) => <svg {...props} />} />,
    );

    expect(getByTestId('drawer-item-icon')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
