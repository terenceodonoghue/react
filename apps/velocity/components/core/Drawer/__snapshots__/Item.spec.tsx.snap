// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item is unselected by default 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: solid 3px transparent;
  color: #b0bac9;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 56px;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding: 0 0 0 25px;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.6, 1);
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.6, 1);
}

.emotion-0:hover {
  background-color: rgba(46,91,255,0.1);
}

.emotion-1 {
  margin-left: 16px;
  opacity: 1;
  -webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
}

<div>
  <li
    class="emotion-0"
    data-testid="drawer-item"
  >
    <span
      class="emotion-1"
      data-testid="drawer-item-text"
    />
  </li>
</div>
`;

exports[`Item renders an icon 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: solid 3px transparent;
  color: #b0bac9;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 56px;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding: 0 0 0 25px;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.6, 1);
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.6, 1);
}

.emotion-0:hover {
  background-color: rgba(46,91,255,0.1);
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 24px;
  width: 24px;
}

.emotion-2 {
  margin-left: 16px;
  opacity: 1;
  -webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
}

<div>
  <li
    class="emotion-0"
    data-testid="drawer-item"
  >
    <svg
      class="emotion-1"
      color="#bfc5d2"
      data-testid="drawer-item-icon"
    />
    <span
      class="emotion-2"
      data-testid="drawer-item-text"
    />
  </li>
</div>
`;

exports[`Item renders compact variant 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: solid 3px transparent;
  color: #b0bac9;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 56px;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding: 0 0 0 25px;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.6, 1);
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.6, 1);
}

.emotion-0:hover {
  background-color: rgba(46,91,255,0.1);
}

.emotion-1 {
  margin-left: 16px;
  opacity: 0;
  -webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
}

<div>
  <li
    class="emotion-0"
    data-testid="drawer-item"
  >
    <span
      class="emotion-1"
      data-testid="drawer-item-text"
    />
  </li>
</div>
`;

exports[`Item renders content 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: solid 3px transparent;
  color: #b0bac9;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 56px;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding: 0 0 0 25px;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.6, 1);
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.6, 1);
}

.emotion-0:hover {
  background-color: rgba(46,91,255,0.1);
}

.emotion-1 {
  margin-left: 16px;
  opacity: 1;
  -webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
}

<div>
  <li
    class="emotion-0"
    data-testid="drawer-item"
  >
    <span
      class="emotion-1"
      data-testid="drawer-item-text"
    >
      The SMS program is down, synthesize the redundant feed so we can synthesize the SQL hard drive!
    </span>
  </li>
</div>
`;

exports[`Item renders selected item 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(46,91,255,0.1);
  border-left: solid 3px #2e5bff;
  color: #2e5bff;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 56px;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding: 0 0 0 25px;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.6, 1);
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.6, 1);
}

.emotion-1 {
  margin-left: 16px;
  opacity: 1;
  -webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
}

<div>
  <li
    class="emotion-0"
    data-testid="drawer-item"
  >
    <span
      class="emotion-1"
      data-testid="drawer-item-text"
    />
  </li>
</div>
`;

exports[`Item shows text by default 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: solid 3px transparent;
  color: #b0bac9;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 56px;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding: 0 0 0 25px;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.6, 1);
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.6, 1);
}

.emotion-0:hover {
  background-color: rgba(46,91,255,0.1);
}

.emotion-1 {
  margin-left: 16px;
  opacity: 1;
  -webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
}

<div>
  <li
    class="emotion-0"
    data-testid="drawer-item"
  >
    <span
      class="emotion-1"
      data-testid="drawer-item-text"
    />
  </li>
</div>
`;
